components/tidb_query/src/rpn_expr/impl_miscellaneous.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use tidb_query_codegen::rpn_fn;
+
+use crate::codec::data_type::*;
+use crate::Result;
+
+use std::net::Ipv4Addr;
+
+#[rpn_fn]
+#[inline]
+pub fn inet_ntoa(arg: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(match arg {
+        Some(arg) => {
+            if *arg < 0 || *arg > i64::from(u32::max_value()) {
+                None
+            } else {
+                let v = *arg as u32;
+                let ipv4_addr =
+                    Ipv4Addr::new((v >> 24) as u8, (v >> 16) as u8, (v >> 8) as u8, v as u8);
```

Comment:
Is it necessary to convert it to `Ipv4Addr` first? Why not construct the output string directly?

Reply:
Another option is using `From<u32>` to create `Ipv4Addr` and formating it to string